src/app/identity.service.ts

```diff
@@ -0,0 +1,75 @@
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+import { Inject, Injectable, InjectionToken } from '@angular/core';
+import { TransferState } from '@angular/platform-browser';
+import { StorageService } from 'ngx-webstorage-service';
+import { Observable } from 'rxjs';
+
+import { AnalyticsService } from './analytics.service';
+import { environment } from '../environments/environment';
+import { Person } from './person.model';
+
+export const TBG_DONATE_ID_STORAGE = new InjectionToken<StorageService>('TBG_DONATE_ID_STORAGE');
+
+@Injectable({
+  providedIn: 'root',
+})
+export class IdentityService {
+  private readonly apiPath = '/people';
+  // Key is per-domain/env. For now we simply store a single JWT (or none).
+  private readonly storageKey = `${environment.identityApiPrefix}/v1/jwt`;
+
+  constructor(
+    private analyticsService: AnalyticsService,
+    private http: HttpClient,
+    @Inject(TBG_DONATE_ID_STORAGE) private storage: StorageService,
+  ) {}
+
+  create(person: Person): Observable<Person> {
+    return this.http.post<Person>(
+      `${environment.identityApiPrefix}${this.apiPath}`,
+      person);
+  }
+
+  update(person: Person): Observable<Person> {
+    return this.http.put<Person>(
+      `${environment.identityApiPrefix}${this.apiPath}/${person.id}`,
+      person,
+      this.getAuthHttpOptions(person),
+    );
+  }
+
+  getIdAndJWT(): { id: string, jwt: string } | undefined {
+    if (!this.storage.has(this.storageKey)) {
+      return undefined;
+    }
+
+    return this.storage.get(this.storageKey) || undefined;
+  }
+
+  getJWT(): string | undefined {
+    return this.getIdAndJWT()?.jwt || undefined;
```

Comment:
`|| undefined` here seems like unreachable code if `this.getIdAndJWT()?.jwt` can already return `undefined`